state/indexer/indexer_service.go

```diff
@@ -75,25 +76,34 @@ func (is *Service) OnStart() error {
 				}
 			}
 
-			if err := is.blockIdxr.Index(eventDataHeader); err != nil {
-				is.Logger.Error("failed to index block", "height", height, "err", err)
-			} else {
-				is.Logger.Info("indexed block", "height", height)
-			}
+			for _, sink := range is.eventSinks {
+				if err := sink.IndexBlockEvents(eventDataHeader); err != nil {
+					is.Logger.Error("failed to index block", "height", height, "err", err)
+				} else {
+					is.Logger.Info("indexed block", "height", height)
+				}
 
-			if err = is.txIdxr.AddBatch(batch); err != nil {
-				is.Logger.Error("failed to index block txs", "height", height, "err", err)
-			} else {
-				is.Logger.Debug("indexed block txs", "height", height, "num_txs", eventDataHeader.NumTxs)
+				for _, result := range batch.Ops {
+					err := sink.IndexTxEvents(result)
+					if err != nil {
+						is.Logger.Error("failed to index block txs", "height", height, "err", err)
+					}
+				}
 			}
 		}
 	}()
 	return nil
 }
 
-// OnStop implements service.Service by unsubscribing from all transactions.
+// OnStop implements service.Service by unsubscribing from all transactions and
+// close the sqlDB if the service is using the psqlEventSink
 func (is *Service) OnStop() {
 	if is.eventBus.IsRunning() {
 		_ = is.eventBus.UnsubscribeAll(context.Background(), subscriber)
 	}
+
+	if is.sqlDB != nil {
```

Comment:
Instead of having a db reference on `Service` which is a codesmell, instead we should consider doing this in `node.go` or having `EventSink` have a `Cleanup` or `Stop` method.

Reply:
@alexanderbez Thanks, so I can add a function in the eventsink interface? I was thinking it is not aligned with the ADR, which might cause confusion in the future implementation.

Reply:
Yes, add a method to the `EventSink` interface 👍